packages/presentation/lib/src/analitics/service.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:firebase_analytics/firebase_analytics.dart';
+
+class Analytics {
+  final FirebaseAnalytics firebaseAnalytics;
+
+  const Analytics(
+    this.firebaseAnalytics,
+  );
+
+  logOnGoogleAuthClick() =>
+      firebaseAnalytics.logEvent(name: 'AuthWithGoogle clicked');
+
+  logOnFacebookAuthClick() =>
+      firebaseAnalytics.logEvent(name: 'AuthWithFacebook clicked');
+
+  logWithEmailAndPassClick() =>
+      firebaseAnalytics.logEvent(name: 'AuthWithEmailAndPassword clicked');
```

Comment:
no need in creating separate method for each event. in this case our service will be too big. just create one method - logEvent and pass here different params